app/questionnaire/dependencies.py

```diff
@@ -0,0 +1,39 @@
+from collections import defaultdict
+
+
+class Dependencies:
+    """represents the dependencies for a group of answers
+    held internally as a dict(string,set) where set contains dependent ids
+    set used to prevent duplicates
+
+    To add a dependency either call add with a single dependency
+    or a+=b where both a and b are Dependencies objects.
```

Comment:
I'm not sure it's a good idea to implement the `+` operator for the `Dependencies` object when the `Dependencies` object itself is largely acting as a dict and it expects a dict to be added to it. Dicts can't be added together, that's what `update()` is for, so I'd suggest implementing that instead.